### DISCUSSION TITLE
{{#if product.can_purchase}} Not Working on the Product Card

### DISCUSSION THREAD

**Discussion by Mj6078** [community/community#140409] [Programming Help]
### Select Topic Area

Question

### Body

    {{#if product.can_purchase}}
      <button
        type="submit"
        class="
          button
          button-primary
          button-wide
          add-to-cart
          button-progress
          spinner
        "
        data-button-purchase
        data-event-type="product-click"
        {{#unless product.can_purchase}}
          disabled
        {{/unless}}
      >
        <span class="button-text">
          {{#unless product.can_purchase}}
            {{lang 'product.sold_out'}}
          {{else}}
            {{#if product.pre_order}}
              {{lang 'product.pre_order'}}
            {{else if cart_icon '==' 'bag'}}
              {{lang 'product.add_to_bag'}}
            {{else}}
              {{lang 'product.add_to_cart'}}
            {{/if}}
          {{/unless}}
        </span>
      </button>
    {{/if}}

This above code is added to the cart button which is by default on the product page and on there this button is working accurately and perfectly but when I try to put these same conditions on the product card only {{#if product.can_purchase}} This condition is not working. can anyone please explain to me why these conditions are not working?

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Mj6078 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by Mj6078** [reply]
Thank you for choosing The Right category For me.
